src/map/navigation_container.cpp

```diff
@@ -0,0 +1,924 @@
+                    if (m_nsmoothpath < Navigation::MAX_SMOOTH)
+                    {
+                        dtVcopy(&m_smoothPath[m_nsmoothpath * 3], iterPos);
+                        m_nsmoothpath++;
+                    }
+                    break;
+                }
+                else if (offMeshConnection && Navigation::dtInRange(iterPos, steerPos, Navigation::PATH_SLOP, 1.0f))
+                {
+                    float startPos[3], endPos[3];
+
+                    dtPolyRef prefRef = 0, polyRef = polys[0];
+                    int       npos = 0;
+                    while (npos < npolys && polyRef != steerPosRef)
+                    {
+                        prefRef = polyRef;
+                        polyRef = polys[npos];
+                        npos++;
+                    }
+                    for (int i = npos; i < npolys; ++i)
+                    {
+                        polys[i - npos] = polys[i];
+                    }
+                    npolys -= npos;
+
+                    dtStatus status = m_navMesh->getOffMeshConnectionPolyEndPoints(prefRef, polyRef, startPos, endPos);
+                    if (dtStatusSucceed(status))
+                    {
+                        if (m_nsmoothpath < Navigation::MAX_SMOOTH)
+                        {
+                            dtVcopy(&m_smoothPath[m_nsmoothpath * 3], startPos);
+                            m_nsmoothpath++;
+                            if (m_nsmoothpath & 1)
+                            {
+                                dtVcopy(&m_smoothPath[m_nsmoothpath * 3], startPos);
+                                m_nsmoothpath++;
+                            }
+                        }
+
+                        dtVcopy(iterPos, endPos);
+                        float eh = 0.0f;
+                        m_query.getPolyHeight(polys[0], iterPos, &eh);
+                        iterPos[1] = eh;
+                    }
+                }
+
+                // Store results
+                if (m_nsmoothpath < Navigation::MAX_SMOOTH)
+                {
+                    dtVcopy(&m_smoothPath[m_nsmoothpath * 3], iterPos);
+                    m_nsmoothpath++;
+                }
+            }
+        }
+    }
+    else
+    {
+        m_npolys      = 0;
+        m_nsmoothpath = 0;
+    }
+
+    if (m_nsmoothpath > 0)
+    {
+        for (int i = 3; i < m_nsmoothpath * 3;)
+        {
+            float pathPos[3];
+            pathPos[0] = m_smoothPath[i++];
+            pathPos[1] = m_smoothPath[i++];
+            pathPos[2] = m_smoothPath[i++];
+            Navigation::detourToGamePosition(pathPos);
+            m_points.push_back({ { pathPos[0], pathPos[1], pathPos[2], 0, 0 }, 0, 0 });
+        }
+    }
+
+    return m_points;
+}
+
+auto CNavigationContainer::findRandomPosition(const position_t& t_around, float t_radius,
+                                              unsigned short includePolyFlags, unsigned short excludePolyFlags)
+    -> pathpoint_t
+{
```

Comment:
I gotta say, I'm not on board for the trailing return type style in the function definition. Especially if clang format tries to indent the trailing return type. It looks and works a lot better in the declaration, but not here.